src/main/java/org/opensearch/ad/InsightsJobProcessor.java

```diff
@@ -0,0 +1,666 @@
+                    AnomalyResult anomaly = AnomalyResult.parse(parser);
+                    allAnomalies.add(anomaly);
+                } catch (Exception e) {
+                    log.warn("Failed to parse anomaly from {} (document may be incomplete)", hit.getId(), e);
+                }
+            }
+
+            // when search results is less than one page
+            if (hits.length == 0 || hits.length < baseSource.size()) {
+                log
+                    .info(
+                        "Successfully parsed {} anomalies in time window {} to {}",
+                        allAnomalies.size(),
+                        executionStartTime,
+                        executionEndTime
+                    );
+
+                if (!allAnomalies.isEmpty()) {
+                    // Enrich detector metadata (names, indices) before correlation
+                    fetchDetectorMetadataAndProceed(allAnomalies, jobParameter, lockService, lock, executionStartTime, executionEndTime);
+                } else {
+                    log.info("No anomalies found in time window, skipping ML correlation");
+                    releaseLock(jobParameter, lockService, lock);
+                }
+                return;
+            }
+
+            // continue to next page
+            Object[] next = hits[hits.length - 1].getSortValues();
+            fetchPagedAnomalies(
+                baseSource,
+                next,
+                allAnomalies,
+                jobParameter,
+                lockService,
+                lock,
+                executionStartTime,
+                executionEndTime
+            );
+        }, e -> {
+            if (e.getMessage() != null && (e.getMessage().contains("no such index") || e.getMessage().contains("index_not_found"))) {
+                log.info("Anomaly results index does not exist yet (no anomalies recorded)");
+            } else {
+                log.error("Failed to query anomaly results", e);
+            }
+            releaseLock(jobParameter, lockService, lock);
+        }));
+    }
+
+    /**
+     * Process anomalies with ML Commons metrics correlation.
+     * 
+     * @param jobParameter The insights job
+     * @param lockService Lock service for releasing lock
+     * @param lock The acquired lock
+     * @param anomalies All collected anomalies
+     * @param detectorMetadataMap Detector metadata for insights generation
+     * @param executionStartTime Start of analysis window
+     * @param executionEndTime End of analysis window
+     */
+    private void processAnomaliesWithMLCommons(
+        Job jobParameter,
+        LockService lockService,
+        LockModel lock,
+        List<AnomalyResult> anomalies,
+        Map<String, DetectorMetadata> detectorMetadataMap,
+        Instant executionStartTime,
+        Instant executionEndTime
+    ) {
+        MLMetricsCorrelationInput input = MLMetricsCorrelationInputBuilder
```

Comment:
Can you have a max limit of the number of time series sent to ML-commons? How about 10 MB? If there are more, you can switch to use Pearson as it only need anomaly points instead of a series.